tests/expectations/constant.compat.c

```diff
@@ -21,8 +21,16 @@
 
 #define QUOTE '\''
 
+#define SHIFT 3
+
 #define TAB '\t'
 
+#define XBOOL 1
+
+#define XFALSE (0 << SHIFT) | XBOOL
+
+#define XTRUE (1 << SHIFT) | XBOOL
```

Comment:
This is pretty footgunny C code. We should probably parenthesize all binary ops when emmitting to a macro. That would also avoid the need for `Literal::Paren`, I think, which would maybe make the patch a bit simpler.

Reply:
@emilio 

> This is pretty footgunny C code. We should probably parenthesize all binary ops when emmitting to a macro. 

I agree, this can go wrong in so many ways - happy to add a parenthesis around the whole expression if there is a binary op present. Should be able to peek into the structure inside the write function do this. 

>  That would also avoid the need for Literal::Paren, I think, which would maybe make the patch a bit simpler.

I agree, adding that variant did make me cringe a bit. I prefer to not have it there. 

Do you think I should just recursively call the value inside and avoid the wrapping all together? Something like this? 

```patch
- syn::Expr::Paren(syn::ExprParen { ref expr, .. }) => Ok(Literal::Paren {
-     inner: Box::new(Self::load(expr)?),
- }),

+ syn::Expr::Paren(syn::ExprParen { ref expr, .. }) => Self::load(expr),
```

Reply:
Yeah, thtat should work, but let me know if it doesn't.

Reply:
I was slightly sceptical but this seems to have worked fine, working on a few tests now to make sure the config and renaming work correctly.